File: leetcode/9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        reversed = 0
        temp = x
        while temp != 0:
            last_digit = temp % 10
            reversed = reversed * 10 + last_digit
            temp //= 10
        return reversed == x

File: leetcode/9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0) {
            return false;
        }
        int old = x;
        long long result = 0;
        while (x!=0) {
            int temp = x % 10;
            result = result * 10 + temp;
            x = x / 10;
        }
        if (old == int(result)) {
            return true;
        }
        return false;
    }
};